classes/search/block/query/class-parameters.php

```diff
@@ -167,6 +173,13 @@ public function post_status(): ?array {
 		return $this->post_status ? $this->post_status : self::DEFAULT_POST_STATUS;
 	}
 
+	/**
+	 * List of required post types.
+	 */
+	public function post_type(): ?array {
+		return $this->post_type ? $this->post_type : null;
```

Comment:
What do you think if we return `$this->post_type ?? null;`?

Reply:
Also `post_type` is defined [here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/871/files#diff-fea718b863cbbd69a7354610f017c389dae8078205f63da9d69ca366792b5b3bR42) as a `string` but this function returns an `array` or a nully value. Am I right?

Reply:
haha yes sorry, I think the PHPCS rules refuse `??` at the moment so I end up with stuff like that :facepalm: I'll change the rule instead of complying blindly :pray:

Reply:
Ha, so in fact post_type is an array of strings to holds multiple types, like post_status holds multiple statuses; the declaration `string[]` means an array `[]` of `string`, it's a way to [specify the content of the array](https://docs.phpdoc.org/3.0/guide/references/phpdoc/types.html#arrays) 
There are [other type declarations](https://phpstan.org/writing-php-code/phpdoc-types#general-arrays) that we can use once we plug something like phpstan on the code.